pinot-perf/src/main/java/org/apache/pinot/perf/BenchmarkStringDictionaryUnPaddedString.java

```diff
@@ -0,0 +1,142 @@
+  private static final Random RANDOM = new Random();
+  private static final int MAX_DOC_PER_CALL = DocIdSetPlanNode.MAX_DOC_PER_CALL;
+  private static final int BYTES_PER_VALUE = 36; // UUID string length
+  private static final String TABLE_NAME = "tbl1";
+  private static final String filePath = "/home/user/benchmarks/uuid.bin";
+  private static final int[] dictIds = new int[MAX_DOC_PER_CALL];
+
+  private PinotDataBuffer _dataBuffer;
+
+  @Setup
+  public void setUp()
+      throws Exception {
+    _dataBuffer = PinotDataBuffer.mapFile(new File(filePath), true, 0, 36000000, ByteOrder.BIG_ENDIAN, null);
+    for (int i = 0; i < dictIds.length; i++) {
+      dictIds[i] = RANDOM.nextInt(1_000_000);
+    }
+  }
+
+  @TearDown
+  public void tearDown()
+      throws Exception {
+    _dataBuffer.close();
+  }
+
+  public static void createUUIDFile() {
+    final int bytesPerValue = 36;
+    final int numUUIDs = 1_000_000;
+    byte[] values = new byte[numUUIDs * bytesPerValue];
+    for (int i = 0; i < numUUIDs; i++) {
+      String uuid = UUID.randomUUID().toString();
+      byte[] uuidBytes = uuid.getBytes(StandardCharsets.UTF_8);
+      System.arraycopy(uuidBytes, 0, values, i * bytesPerValue, uuidBytes.length);
+    }
+    // Write values to a file: /home/user/benchmarks/uuid.bin. Don't use PinotDataBuffer.
+    String filePath = "/home/user/benchmarks/uuid.bin";
+    try (FileOutputStream fileOutputStream = new FileOutputStream(filePath)) {
+      fileOutputStream.write(values);
+    } catch (Exception e) {
+      throw new RuntimeException("Failed to write UUIDs to file", e);
+    }
+  }
+
+  @Benchmark
+  public void benchmarkUnpaddedStringDefaultApproach() {
+    FixedByteValueReaderWriter reader = new FixedByteValueReaderWriter(_dataBuffer);
+    byte[] uuidBytes = new byte[BYTES_PER_VALUE];
+    String[] result = new String[MAX_DOC_PER_CALL];
+    for (int index = 0; index < dictIds.length; index++) {
+      int dictId = dictIds[index];
+      reader.readUnpaddedBytes(dictId, BYTES_PER_VALUE, uuidBytes);
+      result[index] = new String(uuidBytes, StandardCharsets.UTF_8);
+      // result[index] = reader.getPaddedString(dictId, BYTES_PER_VALUE, uuidBytes);
+    }
+  }
+
+  @Benchmark
+  public void benchmarkGetPaddedString() {
+    FixedByteValueReaderWriter reader = new FixedByteValueReaderWriter(_dataBuffer);
+    byte[] uuidBytes = new byte[BYTES_PER_VALUE];
+    String[] result = new String[MAX_DOC_PER_CALL];
+    for (int index = 0; index < dictIds.length; index++) {
+      int dictId = dictIds[index];
+      result[index] = reader.getPaddedString(dictId, BYTES_PER_VALUE, uuidBytes);
+    }
+  }
+
+  @Benchmark
+  public void benchmarkGetCustomPaddedString() {
+    FixedByteValueReaderWriter reader = new FixedByteValueReaderWriter(_dataBuffer);
+    byte[] uuidBytes = new byte[BYTES_PER_VALUE];
+    String[] result = new String[MAX_DOC_PER_CALL];
+    ByteBuffer byteBuffer = ByteBuffer.wrap(uuidBytes);
```

Comment:
this buffer is in little endian but Pinot index files are in big endian. so the numbers shared in PR description don't matter